internal/feather.go

```diff
@@ -315,7 +323,56 @@ func ColumnMapToRecord(inputs map[string]any) (arrow.Record, error) {
 			return nil, errors.Wrapf(err, "failed to set values for feature '%s'", field.Name)
 		}
 	}
-	return recordBuilder.NewRecord(), nil
+
+	res := recordBuilder.NewRecord()
+
+	// Need to convert __ts__ to exactly pa.timestamp("us", "UTC")
+	// until the UploadFeatures gRPC endpoint handles precisions
+	// other than "us".
+	tsIndex := -1
+	var tsField arrow.Field
+	for idx, field := range res.Schema().Fields() {
+		if field.Name == "__ts__" {
+			tsIndex = idx
+			tsField = field
+			break
+		}
+	}
+	if tsIndex != -1 {
+		defer res.Release()
+		tsBuilder := array.NewTimestampBuilder(allocator, &arrow.TimestampType{Unit: arrow.Microsecond, TimeZone: "UTC"})
+		defer tsBuilder.Release()
+		tsValues := make([]arrow.Timestamp, res.NumRows())
+		validityMask := make([]bool, res.NumRows())
+		for i := 0; i < int(res.NumRows()); i++ {
+			validityMask[i] = res.Column(tsIndex).IsValid(i)
+			tsCol, ok := res.Column(tsIndex).(*array.Timestamp)
+			if !ok {
+				return nil, fmt.Errorf("failed to cast '__ts__' column to Timestamp Array")
+			}
+			val := tsCol.Value(i)
```

Comment:
is it valid to get the value if isValid() return false? Or is it expected that all invalid values are zero-constructed?

Reply:
Yes - unless the method returns an error, it shouldn't panic/go into an error state in any case

Reply:
but good caution